frontend/src/util/index.ts

```diff
@@ -234,3 +240,68 @@ export const secondsToTimeString = (seconds: number): string => {
 };
 
 export type ExtraMetadata = Record<string, Record<string, string[]>>;
+
+export type Credentials = {
+    user: string;
+    password: string;
+} | null;
+
+interface AuthenticatedData extends OperationType {
+    response: VideoAuthorizedDataQuery$data;
+}
+
+/**
+ * Returns `authorizedData` of password protected events by fetching it from the API,
+ * if the correct credentials were supplied.
+ * This will not send a request when there are no credentials and instead return the
+ * event's authorized data if that was already present and passed to this hook.
+ *
+ * ===ETH SPECIAL FEATURE===
+ * This can be passed a series id and the authenticated data of an event.
+ * If the `interpret_eth_passwords` feature is enabled, this will check if credentials
+ * with that reference are stored and, if they are and the authenticated data is not yet
+ * present, query and return that data.
+ */
+export const useAuthenticatedDataQuery = (
+    eventID: string,
+    seriesID?: string,
+    authData?: AuthorizedData | null,
+) => {
+    // If `id` is coming from a search event, the prefix might be `es` or `ss`, but
+    // the query and storage need it to be an event/series id (i.e. with prefix `ev`/`sr`).
+    const credentials = seriesID && CONFIG.sync.interpretETHPasswords
+        ? getCredentials(seriesId(keyOfId(seriesID)))
+        : getCredentials(eventID);
+    const authenticatedData = useLazyLoadQuery<AuthenticatedData>(
+        authorizedDataQuery,
+        { eventId: eventId(keyOfId(eventID)), ...credentials },
+        // This will only query the data for events with stored credentials and/or yet unknown
+        // authorized data. This should help to prevent unnecessary queries.
+        { fetchPolicy: !(credentials || authData) ? "store-only" : "store-or-network" }
```

Comment:
```suggestion
        { fetchPolicy: (credentials || authData) ? "store-or-network" : "store-only" }
```

Seems more straight forward to me. Which also makes me notice: is the logic wrong? If `authData` is not null, then we *dont* want to fetch from network, right? :thinking: